app/APIs/SourceLeecher/SourceFactory.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: macintosh
+ * Date: 11/22/15
+ * Time: 9:58 PM
+ */
+
+namespace App\APIs\SourceLeecher;
+
+
+class SourceFactory implements ISource
+{
+    /**
+     * @ISource sourceObject
+     */
+    private $sourceObject;
+
+    public function __construct($sourceName) {
+        $classString = __NAMESPACE__."\\Sources\\$sourceName";
+
+        try {
+            $this->sourceObject = new $classString();
+        } catch(\Exception $e) {
+            throw new \Exception('Source does not exist', 99991);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS112)